### PR TITLE
Attempts to fetch workspace from Master if all slaves are offline

### PR DESCRIPTION
**Pull request by alokjoshi89**

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java [hunk 1]
```diff
@@ -21,20 +21,25 @@
...
-                List<PodTemplate> podTemplates = kubernetesCloud.getTemplates();
-                for(PodTemplate podTemplate : podTemplates) {
-                    if(ObjectUtils.equals(podTemplate.getLabel(), assignedLabel)) {
-                        String workspacePath = Jenkins.getInstance().getWorkspaceFor((TopLevelItem) job).toString();
-                        LOGGER.info("Workspace Path: " + workspacePath);
-                        File workspace = new File(workspacePath);
-                        LOGGER.info("Workspace exists ? " + workspace.exists());
-                        if (workspace.exists()) {
-                            return new FilePath(workspace);
+        if(job != null) {
```
Comment (iocanel) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: There is no need to add a null check before an `instanceof`.
http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2

Comment (alokjoshi89) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: This makes sense. I'll change it.

Comment (alokjoshi89) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: Where should add the comment about this in readme? I am not sure about the proper location but I was thinking at the end of the file with header 'Workspace support from Master' and details in the body?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public class KubernetesWorkspaceBrowser extends WorkspaceBrowser{
+
+    private static final Logger LOGGER = Logger.getLogger(KubernetesWorkspaceBrowser.class.getName());
+
+    @Override
+    public FilePath getWorkspace(Job job) {
+        LOGGER.info("Nodes went offline. Hence fetching it through master");
+        if (job instanceof AbstractProject) {
+            String assignedLabel = ((AbstractProject) job).getAssignedLabelString();
+            KubernetesCloud kubernetesCloud = Hudson.getInstance().clouds.get(KubernetesCloud.class);
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: You can have multiple KubernetesClouds, each with a different id

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            if (kubernetesCloud != null) {
+                List<PodTemplate> podTemplates = kubernetesCloud.getTemplates();
+                for (PodTemplate podTemplate : podTemplates) {
+                    if (ObjectUtils.equals(podTemplate.getLabel(), assignedLabel)) {
+                        FilePath filePath = Jenkins.getInstance().getWorkspaceFor((TopLevelItem) job);
+                        if(filePath != null) {
+                            String workspacePath = filePath.toString();
+                            LOGGER.info("Workspace Path: " + workspacePath);
+                            File workspace = new File(workspacePath);
+                            LOGGER.info("Workspace exists ? " + workspace.exists());
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: These logging statements should be level `FINE`

**Comment by alokjoshi89** [comment]
Closing this PR as I messed up part of the code. Will create a new one.

**Pull request closed without merging**